xarray/core/dataset.py

```diff
@@ -6403,7 +6406,14 @@ def to_dask_dataframe(
         columns.extend(k for k in self.coords if k not in self.dims)
         columns.extend(self.data_vars)
 
+        has_many_dims = len(ordered_dims) > 1
```

Comment:
Can we move this to a separate PR please?

Reply:
See,  #7844

Reply:
Done.